src/tag_index.c

```diff
@@ -424,3 +424,44 @@ int TagIndex_RegisterType(RedisModuleCtx *ctx) {
 
   return REDISMODULE_OK;
 }
+
+// static size_t TagIndex_GetSuffixOverhead(TagIndex *idx) {
+//   if (!idx->suffix) {
+//     return 0;
+//   }
+//   size_t overhead = 0;
+//   TrieMapIterator *it = TrieMap_Iterate(idx->suffix, "", 0);
+//   char *str;
+//   tm_len_t slen;
+//   void *val;
+//   while (TrieMapIterator_Next(it, &str, &slen, &val)) {
+//     arrayof(char *) arr = ((suffixData *)val)->array;
+//     // Count the size of the array
+//     overhead += array_sizeof(array_hdr(arr));
+//     // Count the term only if allocated
+//     if (((suffixData *)val)->term) {
+//       overhead += slen;
+//     }
+//   }
+
+//   return overhead;
+// }
+
+size_t TagIndex_GetOverhead(IndexSpec *sp, FieldSpec *fs) {
+  size_t overhead = 0;
+  TagIndex *idx = NULL;
+  RedisSearchCtx sctx = SEARCH_CTX_STATIC(RSDummyContext, sp);
+  RedisModuleString *keyName = TagIndex_FormatName(&sctx, fs->name);
+  idx = TagIndex_Open(&sctx, keyName, 0, NULL);
+  RedisModule_FreeString(RSDummyContext, keyName);
+  if (idx) {
+    overhead = TrieMap_MemUsage(idx->values);     // Values' size are counted in stats.invertedSize
+    if (idx->suffix) {
+      overhead += TrieMap_MemUsage(idx->suffix);
+      // We currently avoid counting the suffix values' overhead for performance.
```

Comment:
Can be in separate PR?

Reply:
Why separate PR? This is used.
Do you mean the comment??